File: daspro/tambah_arr.cpp
#include <stdio.h>
#include <string.h>

int main() {
    // Daftar nama-nama buah
    char buah[6][10] = {"apel", "anggur", "mangga", "Jeruk"};
    int jumlahBuah = 4;  // Jumlah elemen awal dalam array

    // Menambahkan buah
    if (jumlahBuah < 6) {
        // Geser elemen ke kanan
        for (int i = jumlahBuah; i > 2; i--) {
            for (int j = 0; j < 10; j++) {
                buah[i][j] = buah[i - 1][j];
            }
        }

        // Menambahkan "duren" di indeks ke-2
        for (int i = 0; i < 10; i++) {
            buah[2][i] = "duren"[i];
        }
        jumlahBuah++;  // Update jumlah buah
    }

    // Tampilkan semua daftar buah
    printf("Semua Buah: ada %d macam\n", jumlahBuah);
    for (int i = 0; i < jumlahBuah; i++) {
        printf("%s\n", buah[i]);
    }

    return 0;
}

File: daspro/tambah_arr.pas
program TambahArray;
uses crt;
var
    buah: array[0..5] of string[10]; // Array dengan panjang 6 elemen
    jumlahBuah, i: Integer;
begin
    clrscr;
    // Inisialisasi array dengan beberapa buah
    buah[0] := 'apel';
    buah[1] := 'anggur';
    buah[2] := 'mangga';
    buah[3] := 'Jeruk';

    jumlahBuah := 4; // Jumlah buah awal

    // Menambah buah array
    if jumlahBuah < 6 then
    begin
        for i := jumlahBuah downto 2 do
            buah[i] := buah[i-1]; // Geser elemen ke kanan
        buah[2] := 'durian'; // Menambahkan "durian" di indeks 2
        jumlahBuah := jumlahBuah + 1; // Update jumlah buah
    end;

    // Tampilkan semua daftar buah
    writeln('Semua Buah: ada ', jumlahBuah, ' macam');
    for i := 0 to jumlahBuah - 1 do
        writeln(buah[i]);
    readln;
end.
